WordPress/Classes/ViewRelated/Post/Prepublishing/PrepublishingSocialAccountsViewController.swift

```diff
@@ -172,6 +228,52 @@ private extension PrepublishingSocialAccountsViewController {
         self.navigationController?.pushViewController(multiTextViewController, animated: true)
     }
 
+    func subscribeButtonTapped() {
+        guard let checkoutViewController = makeCheckoutViewController() else {
+            return
+        }
+
+        let navigationController = UINavigationController(rootViewController: checkoutViewController)
+        show(navigationController, sender: nil)
+    }
+
+    func makeCheckoutViewController() -> UIViewController? {
+        coreDataStack.performQuery { [weak self] context in
+            guard let self,
+                  let blog = try? Blog.lookup(withID: self.blogID, in: context),
+                  let host = blog.hostname,
+                  let url = URL(string: "https://wordpress.com/checkout/\(host)/jetpack_social_basic_yearly") else {
+                return nil
+            }
+
+            return WebViewControllerFactory.controller(url: url, blog: blog, source: Constants.webViewSource) {
+                self.onCheckoutDismissed()
+            }
+        }
+    }
+
+    /// When the checkout web view is dismissed, try to sync the latest sharing limit in case the user did make
+    /// a purchase. We can make this assumption if the returned `sharingLimit` is nil, which means there's no longer
+    /// any sharing limit for the site.
+    func onCheckoutDismissed() {
+        service.syncSharingLimit(for: blogID) { [weak self] result in
+            switch result {
+            case .success(let sharingLimit):
+                self?.sharingLimit = sharingLimit
+                self?.toggleInteractivityIfNeeded()
+
+                if sharingLimit == nil {
+                    self?.tableView.reloadData()
+                }
+
+                // TODO: Inform prepublishing VC to reload data.
+
+            default:
+                break
+            }
+        }
+    }
```

Comment:
Ah, you're right. Good catch! I've updated this in https://github.com/wordpress-mobile/WordPress-iOS/pull/21179/commits/fbd98d9e1e68f6f07b1fd39736f12ece55133e4b to sync the blog instead of the sharing limit information.